WooCommerce/Classes/ViewRelated/Products/UpdateInventory/UpdateProductInventoryViewModel.swift

```diff
@@ -180,9 +217,17 @@ private extension UpdateProductInventoryViewModel {
         isPrimaryButtonLoading = true
 
         // TODO: Handle error
-        try? await inventoryItem.updateStockQuantity(with: newQuantity, stores: stores)
+        do {
+            inventoryItem = try await inventoryItem.updateStockQuantity(with: newQuantity, stores: stores)
 
-        isPrimaryButtonLoading = false
-        updateQuantityButtonMode = .increaseOnce
+            isPrimaryButtonLoading = false
+            updateQuantityButtonMode = .increaseOnce
+        }
+        catch {}
+    }
+
+    func refresh() {
+        viewMode = inventoryItem.manageStock ? .stockCanBeManaged : .stockManagementNeedsToBeEnabled
+        quantity = inventoryItem.stockQuantity?.formatted() ?? ""
```

Comment:
Is this a method we'd like to move to `InventoryItem`? so we only need to call something like `inventoryItem.formattedStock()`

Reply:
Good suggestion @iamgabrielma! As this is just one line, and the logic is the same for all the protocol implementations, I'd prefer to leave it as it is., instead of having to create extra functions on the protocol and its implementations.